data/transport missions.txt

```diff
@@ -1431,6 +1434,241 @@ mission "Hallucination"
+		"rejected illegal jobs" --
+
+	on decline
+		"rejected illegal jobs" ++
+
+	on enter
+		conversation
+			`You'd begun to doubt the wisdom of your decision to transport this band of cutthroats from the moment you agreed to it, but the money was too good to resist. You settle into the captain's chair and prepare for the voyage ahead.`
+			`	But something feels wrong. You expect to hear the sounds of drunkenness and violence coming from the passenger compartment, but instead a deathly silence fills the air. You whirl around in your seat and come face-to-face with the entire gang. They're all armed. And their weapons are pointing at you. Your heart sinks.`
+			`	Their leader appears to be a lithe and dangerous-looking young woman who holds an enormous shotgun that's almost as big as her. "Looks like we boarded the ship of fools, and here's the biggest one!" she says mockingly. "We didn't have a credit to our names, but soon we'll own your ship. Hand over the access codes and we'll let you live."`
+			choice
+				`(Shoot first!)`
+					goto shoot
+				`(Negotiate.)`
+					goto negotiate
+				`(Surrender.)`
+					goto betrayed
+
+			label shoot
+			branch "shoot and die" "shoot and live"
+				random < 75
+
+			label "shoot and die"
+			`That turns out to be a really bad decision. It's also your last. A truly excessive barrage of projectiles ends your life.`
+				die
+
+			label negotiate
+			`"You don't want my ship, you want my money," you stammer, stalling for time while you try to think of a way out of this situation. "This ship is registered to me; nobody would believe that you came across it honestly. You want a fresh start in an anonymous ship. I'll give you 4 million credits, and you can buy a Modified Argosy or some other vessel better suited to your line of work."`
+			` The gang leader snorts derisively. "Why don't we just kill you and hotwire this ship? That sounds more fun."`
+			choice
+				`(Shoot first!)`
+					goto shoot
+				`(Offer more money.)`
+					goto "more money"
+				`(Surrender.)`
+					goto betrayed
+
+			label "more money"
+			`You continue: "You'll have a heck of a time taking over the ship without casualties. My bank account can't be accessed without my permission, and you know it. Use violence and you'll still be poor - and some of you might be dead. I'll give you 8 million, and that's my final offer."`
+			`	8 million credits is a lot of money. Thankfully, it's enough. A sigh of relief escapes your lips as everyone lowers their weapons.`
+			`	You arrange for a merchant captain to transport the renegades down to the surface, along with a password-protected credit chit. Once the airlock closes, you give the password. A lump appears in your throat as your communicator buzzes, alerting you to an unusually large transaction. At least you have escaped with your life and your ship. You are now free to continue your career--poorer, but hopefully wiser.`
+				defer
```

Comment:
This one isn't; I didn't set `repeat` on it.

Will change the em dash.

Reply:
I know it isn't set to repeat, but defer will typically decline a mission without having triggered the condition to say that it offered, meaning it is able to reoffer in the future.

Reply:
Oh hmm. Any suggestions?

Reply:
It can't be `decline` because at this point the mission is already accepted. The dialog only shows after you accept the mission and take off.

Reply:
I know, that's why I said I'm unsure if `defer` in the `on enter` lets it be offered again. When you test this mission and take the route where it defers, do you have `"<mission name>: offered"` in your save file conditions?

Reply:
For a non-`offer` MissionAction, the callback executed by the conversation exit is different: https://github.com/endless-sky/endless-sky/blob/f8675db93be42c1e179b59291beaee41f37635f5/source/MissionAction.cpp#L390-L401

It's pretty short:
https://github.com/endless-sky/endless-sky/blob/f8675db93be42c1e179b59291beaee41f37635f5/source/PlayerInfo.cpp#L1551-L1557

So `defer` won't do anything different than `accept`, `decline`, `flee`, `launch`, or `depart`. Note that `die` (and `explode`) are handled in the conversation exit, before the callback occurs: https://github.com/endless-sky/endless-sky/blob/f8675db93be42c1e179b59291beaee41f37635f5/source/ConversationPanel.cpp#L345-L371

So the mission will only offer once, because it was accepted and is not `repeat`. You can't change that it was offered or accepted after it was accepted, except through explicit sneakiness: decrementing its offer condition count in an `apply` node.

Reply:
Thanks for the insight.

Reply:
Isn’t `fail` it’s own keyword?

Reply:
No; See https://github.com/endless-sky/endless-sky/issues/1563 and https://github.com/endless-sky/endless-sky/issues/1564

Reply:
Alright then, apply a condition at this point, then add a `to fail` block with `has “<condition>”`. That should do as a work around.